arangod/VocBase/LogicalCollection.cpp

```diff
@@ -790,6 +804,45 @@ arangodb::Result LogicalCollection::properties(velocypack::Slice const& slice,
     }
   }
 
+  if (!minrfSl.isNone()) {
+    if (minrfSl.isInteger()) {
+      int64_t minrfTest = minrfSl.getNumber<int64_t>();
```

Comment:
Just for me: somewhere above you had `getNumber<size_t>`. Here you use `int64_t`. Is this because json only allows signed values and reading unsigned could result in a very huge number? In the second place you made sure, that this value is already validated to be positive?

Reply:
I guess this is just not correct ...

Reply:
Ah, sorry on second thought, it is as done with ReplicationFactor so yes it is because JSON could contain `-1` which is Integer, and getting it as `uint_64t` is incorrect.

Reply:
This would throw for negative values when using `getNumber<uint64_t>()`.

Though `getNumber<int64_t>()` can still throw for some values - e.g. if it contains an unsigned int greater than `INT64_MAX`.

So I'd say we should either allow it to throw in any case, in which case we can just get a `uint64_t` and forget about negativity. Or we do not want to throw, in which case we must catch and thus could also just get a `uint64_t`... But it's not really in the scope of this PR.